src/hooks/useFileUpload.tsx

```diff
@@ -96,6 +100,39 @@ export default function useFileUpload(
   const [files, setFiles] = useState<File[]>([]);
   const queryClient = useQueryClient();
 
+  const generatePDF = async (files: File[]): Promise<File | null> => {
+    try {
+      toast.info(t("file_conversion_in_progress"));
+      const pdf = new jsPDF();
+      const totalFiles = files.length;
+
+      for (const [index, file] of files.entries()) {
+        if (!file.type.startsWith("image/")) {
```

Comment:
With allowedExtensions set, validateFileUpload should be handling this. So not needed here.